docs/customization_extendibility.md

```diff
@@ -0,0 +1,72 @@
+# Extending and Customizing
+For many simpler web applications, `ddev` provides everything you need to successfully provision and develop a web application on your local machine out of the box. More complex and sophisticated web applications, however, often require integration with services beyond the standard requirements of a web and database server. Examples of these additional services are Apache Solr, Redis, Varnish, etc. While `ddev` likely won't ever provide all of these additional services out of the box, it is designed to provide simple ways for the environment to be customized and extended to meet the needs of your application.
```

Comment:
Starting tone here is reasonably more optimistic. Maybe "For most standard web applications"... Paragraph is well written though!